icechunk/src/change_set.rs

```diff
@@ -75,14 +75,16 @@ impl ChangeSet {
         self.new_arrays.get(path)
     }
 
-    pub fn delete_group(&mut self, path: Path, node_id: &NodeId) {
+    pub fn delete_group(&mut self, path: Path, node_id: &NodeId, delete_children: bool) {
```

Comment:
Just pushing this change for discussion. I don't like it. It means an illegal state is representable (remember that we no longer check whether an ancestor of a requested node's path is in `deleted_groups`)

An alternative would be to require that deleting groups goes through `delete_dir` and not through `delete` for icechunk. Any `.delete("/path/to/group/zarr.json")` would fail asking the user to use `delete_dir("/path/to/group")`